### ISSUE TITLE
Roberta classification head output not invariant to padding

### ISSUE THREAD

**Issue by mogeid** [pytorch/fairseq#3628]
I pretrained a Roberta model on my own data.
Now, I've tried registering a new classification head, but its outputs seem to vary with padding. 
Is this expected behaviour or did I make a mistake in the pre-training stage?

#### Code
in this example, `inp[0]` is longer than `inp[1]`, causing `inp[1]` to be padded by `collate`:
```python
coll = collate_tokens(inp[:2], pad_idx=1)
feat = self.roberta.extract_features(coll[1])
padded = self.roberta.model.classification_heads['my_head'].forward(feat)
feat = self.roberta.extract_features(inp[1])
unpadded = self.roberta.model.classification_heads['my_head'].forward(feat)
print(torch.any(unpadded == padded))
# > tensor(False)
```
The same happens when using `predict('my_head, x)` from `hub_interface`.

#### Environnment

 - fairseq 1.0.0
 - PyTorch 1.7
 - Linux

**Comment by knetch** [comment]
Just trying to understand this question myself. But it looks like your tensors are offset by one. Have you tried something like np.isin instead of torch.any?

**Comment by mogeid** [comment]
> Just trying to understand this question myself. But it looks like your tensors are offset by one. Have you tried something like np.isin instead of torch.any?

Well, I guess I'm running into the more basic issue that I'm getting different embeddings from the exact same input
```python
feat1 = self.roberta.extract_features(inp[0])
feat2 = self.roberta.extract_features(inp[0])
print(feat1.shape == feat2.shape)
# > True
print(torch.any(feat1 == feat2))
# > tensor(False)
```
Is there some resetting I am supposed to be doing?

**Comment by mogeid** [comment]
nevermind, I forgot to disable dropout.
